pyomo/contrib/doe/doe.py

```diff
@@ -406,29 +353,28 @@ def compute_FIM(
             The output file uses the name AB.csv, where string A is store_output input, B is the index of scenario.
             scenario index is the number of the scenario outputs which is stored.
         formula:
-            choose from 'central', 'forward', 'backward', None. This option is only used for 'sequential_finite' mode.
+            choose from the Enum finite_difference_step.central, .forward, or .backward.
+            This option is only used for calculation_mode.sequential_finite mode.
         step:
             Sensitivity perturbation step size, a fraction between [0,1]. default is 0.001
-        objective_option:
-            choose from 'det' or 'trace' or 'zero'. Optimization problem maximizes determinant or trace or using 0 as objective function.
 
         Return
         ------
         FIM_analysis: result summary object of this solve
         """
 
         # save inputs in object
-        self.design_values = design_values
-        self.mode = mode
+        self.design_values = self.design_vars.variable_names_value
         self.scale_nominal_param_value = scale_nominal_param_value
         self.scale_constant_value = scale_constant_value
         self.formula = formula
+        self.mode = mode
```

Comment:
This should cast the incoming argument to the enum (so an error is generated if sometthing not in the Enum is passed).  e.g., `self.mode = calculation_mode(mode)`.  Then subsequent references to `mode` should instead reference `self.mode`

Reply:
Enum is updated in this way.